### PR TITLE
Gangs Rework Part 2: Vandalism and Graffiti

### PR DESCRIPTION
**Pull request by TDHooligan**
[GAMEMODES] [FEATURE] [QOL]
## About the PR 
Adds a new feature: **Vandalism**!

image

Occasionally, gangs are tasked with vandalising a department instead of having a duffle drop.
**Vandalism includes:**
 - Graffiti with the new Spray Can
   - A bunch of new graffiti sprites, made by myself and a few community contributors
 - Crowbarring floor tiles (with some prevention)
 - Violence against NPCs
 - Breaking glass tables/Vending machines
 - Breaking lights
 
Gang members must achieve 500 'Vandalism Points' by breaking stuff in the department, where the Duffle bag they're looking for will drop by their feet.

https://github.com/goonstation/goonstation/assets/6061314/68993126-e253-42ab-8cfd-a05f4aa4bd42

## **Additional Features:**
 - Gang members start with 1500 points.
 - Cheap weapons now rebalanced, to compete for your starting points.
 - New graffiti special attack that partially blinds for a few seconds, tagging the victim
 - Baseball bat special now knocks back on crit, with no stuns. Added to gang locker as a middle-tier weapon.
 - New 'Machete' end-game item, competing with D-Saber and Katana
   - New special 'Massacre' attack, with an even more lethal version added to Slasher since Zonespace liked it.
 - 'View scoreboard' button added to the Gang Locker to see where your gang stands
 - Slightly improved locker UI 
 - Spawns have been improved for Duffle Bags. 
 

**Graffiti special:**
'Spray Burst' - A quick, harmless attack that temporarily blinds, defaces and disorients targets.
https://github.com/goonstation/goonstation/assets/6061314/93b8f3ef-b0bf-45fe-a901-9f549b106224

**Baseball special:**
'Baseball Swing' - Swipe, but knocks back on crit (15% chance)

**Machete special:**
'Butcher' - A risky all-in manoeuvre. Three quick low-damage swipes. If they all hit, the user sacrifices all their stamina to continue swinging until they're out of stamina or miss. If they miss, they take a few seconds of disorient.
https://github.com/goonstation/goonstation/assets/6061314/1149a1d5-933b-43fa-b915-776eeda58037

**Machete special - Slasher**
'Massacre' - Similar, but doesn't risk stamina. If the victim doesn't escape, damage escalates, then eventually delimbs, then decapitates.
https://github.com/goonstation/goonstation/assets/6061314/ee67e576-bbdc-47c9-ad7f-1c03769c98c9

## Why's this needed? 
More ways to get duffles reliably. Less dependence on players going along with it.
Also more gang/crew tension. Smashing a department has consequences...
Fixbatch for several QoL issues in there too, and a little more content for more gang variety.

## Changelog 

```changelog
(u)TDHooligan
(*)Gangs may now be tasked with vandalising departments!
(*)Gangs have new weapons & equipment available in their locker.
(*)Gangs can now view a scoreboard from their locker.
(+)Gangs start with 1500 points.
(+)Gangs now have access to regular, cleanable graffiti. This can't be used to claim territory.
(+)Baseball bats' special attack knocks back on crit.
(+)Item costs have been rebalanced to compete for these starting points.
```

### CODE REVIEW

**Comment by TDHooligan** [comment]
graffiti hit sfx still needs implementing
lower graffiti times
telesci fx for the duffle spawning

**Comment by Glamurio** [comment]
This has everything I'd want it to have. It's good that doing the floor graffiti gives significantly more points than simply smashing things, so there's an extra incentive to do that.

Is there any kind of tracker to avoid boring point farming? Like... standing in one spot, and repeatedly breaking the floor tile and fixing it? Does fixing things in a department detract points? If not, maybe it should.

**Comment by TDHooligan** [comment]
boring shit hard to code around and is boring. if its an issue i can fix it.

File: code/modules/status_system/statusEffects.dm [hunk 1]
```diff
@@ -2751,3 +2751,53 @@
 			var/mob/living/M = src.owner
 			M.bioHolder.RemoveEffectInstance(src.added_accent)
 		UnregisterSignal(src.owner, COMSIG_MOB_SAY)
+
+/datum/statusEffect/graffiti
+	id = "graffiti"
+	name = "Tagged!"
+	desc = "You've been tagged! <br>Movement speed is reduced. Eyesight reduced. "
+	icon_state = "tagged"
+	unique = 1
```
Comment (FlameArrow57) [inline code/modules/status_system/statusEffects.dm:2760]:
```suggestion
	unique = TRUE
```

File: code/modules/status_system/statusEffects.dm [hunk 2]
```diff
@@ -2751,3 +2751,53 @@
...
+			var/mob/victim = owner
+			victim.attach_hud(src.hud)
+
+	onRemove()
+		..()
+		if (ismob(owner))
+			var/mob/victim = owner
+			victim.detach_hud(src.hud)
+		for (var/i in 1 to length(tag_images))
+			owner.UpdateOverlays(null,"graffitisplat[i]")
```
Comment (FlameArrow57) [inline code/modules/status_system/statusEffects.dm:2783]:
```suggestion
			owner.ClearSpecificOverlays("graffitisplat[i]")
```
New proc as of recently, I think this is how it works

File: code/modules/status_system/statusEffects.dm [hunk 3]
```diff
@@ -2751,3 +2751,53 @@
...
+		owner.UpdateIcon()
+
+	onUpdate(timePassed)
+		counter += timePassed
+		if (duration < 40)
+			for (var/i in 1 to length(tag_images))
+				var/image/tag = tag_images[i]
+				var/target_alpha = (duration)*5
```
Comment (FlameArrow57) [inline code/modules/status_system/statusEffects.dm:2791]:
```suggestion
				var/target_alpha = duration * 5
```

File: code/modules/status_system/statusEffects.dm [hunk 4]
```diff
@@ -2751,3 +2751,53 @@
...
+				if (tag.alpha > target_alpha)
+					tag.alpha = target_alpha
+					owner.UpdateOverlays(tag,"graffitisplat[i]")
+					owner.UpdateIcon()
+		if (counter >= count && owner && !owner.hasStatus(list("weakened", "paralysis")) )
```
Comment (FlameArrow57) [inline code/modules/status_system/statusEffects.dm:2796]: "weakened" and "paralysis" were renamed to something else recently, don't remember what they are, Leah should know tho

File: code/modules/status_system/statusEffects.dm [hunk 5]
```diff
@@ -2751,3 +2751,53 @@
...
+		if (ismob(owner))
+			var/mob/victim = owner
+			victim.detach_hud(src.hud)
+		for (var/i in 1 to length(tag_images))
+			owner.UpdateOverlays(null,"graffitisplat[i]")
+		owner.UpdateIcon()
+
+	onUpdate(timePassed)
+		counter += timePassed
+		if (duration < 40)
```
Comment (FlameArrow57) [inline code/modules/status_system/statusEffects.dm:2788]:
```suggestion
		if (duration < 4 SECONDS)
```

File: code/modules/power/lighting.dm [hunk 6]
```diff
@@ -1036,6 +1036,11 @@ ADMIN_INTERACT_PROCS(/obj/machinery/light, proc/broken, proc/admin_toggle, proc/
 
 			boutput(user, "You hit the light, and it smashes!")
 			logTheThing(LOG_STATION, user, "smashes a light at [log_loc(src)]")
+
+			var/datum/gang/gang = user.get_gang()
+			if (gang)
+				gang.do_vandalism(GANG_VANDALISM_LIGHT_BREAK_POINTS, src.loc)
```
Comment (FlameArrow57) [inline code/modules/power/lighting.dm:1042]:
```suggestion
			gang?.do_vandalism(GANG_VANDALISM_LIGHT_BREAK_POINTS, src.loc)
```

File: code/modules/items/specials/ItemSpecials.dm [hunk 7]
```diff
@@ -2227,3 +2358,155 @@ ABSTRACT_TYPE(/datum/item_special/spark)
 		if(progress == 1)
 			state = ACTIONSTATE_FINISH
 			return
+
+
+/datum/item_special/massacre
+	cooldown = 20
```
Comment (FlameArrow57) [inline code/modules/items/specials/ItemSpecials.dm:2364]:
```suggestion
	cooldown = 2 SECONDS
```

File: code/modules/items/specials/ItemSpecials.dm [hunk 8]
```diff
@@ -2227,3 +2358,155 @@ ABSTRACT_TYPE(/datum/item_special/spark)
...
+	var/staminacost_chain = 0
+	///Stamina cost increase per extra swing
+	var/staminacost_chain_additive = 5
+	///Disorient and drain stamina when interrupted
+	var/penalty_disorient = TRUE
+	var/alternate = FALSE
+	onAdd()
+		if(master)
+			overrideStaminaDamage = master.stamina_damage * 0.8
+		return
```
Comment (FlameArrow57) [inline code/modules/items/specials/ItemSpecials.dm:2390]:
```suggestion
```

File: code/datums/gamemodes/gangwar.dm [hunk 9]
```diff
@@ -1115,6 +1303,150 @@ proc/broadcast_to_all_gangs(var/message)
...
+		S.charges -= targets
+		S.UpdateIcon()
+		if (targets == 1)
+			var/obj/decal/cleanable/gang_graffiti/tag = new/obj/decal/cleanable/gang_graffiti(S.graffititargets[1])
+			tag.icon_state = iconstate
+			tag.dir = S.tagging_direction
+			if (gang)
+				gang.do_vandalism(GANG_VANDALISM_PER_GRAFFITI_TILE, S.graffititargets[1])
+		else
+			var/list/turf/turfs_ordered = new/list(length(S.graffititargets))
```
Comment (FlameArrow57) [inline code/datums/gamemodes/gangwar.dm:1413]:
```suggestion
			var/list/turf/turfs_ordered = list()
```

File: code/datums/gamemodes/gangwar.dm [hunk 10]
```diff
@@ -1115,6 +1303,150 @@ proc/broadcast_to_all_gangs(var/message)
...
+			var/spraydirection = dir_to_angle(S.tagging_direction)
+			var/vec = angle_to_vector(spraydirection)
+			var/min_distance = 1000
+			for (var/i = 1; i<=targets; i++)
```
Comment (FlameArrow57) [inline code/datums/gamemodes/gangwar.dm:1417]:
```suggestion
			for (var/i = 1 to targets)
```

File: code/datums/gamemodes/gangwar.dm [hunk 11]
```diff
@@ -1115,6 +1303,150 @@ proc/broadcast_to_all_gangs(var/message)
...
+				min_distance = min(vec[2] * S.graffititargets[i].y - vec[1] * S.graffititargets[i].x , min_distance)
+
+			for (var/sorting = 1; sorting<=targets; sorting++)
```
Comment (FlameArrow57) [inline code/datums/gamemodes/gangwar.dm:1420]:
```suggestion
			for (var/sorting = 1 to targets)
```

File: code/datums/gamemodes/gangwar.dm [hunk 12]
```diff
@@ -1115,6 +1303,150 @@ proc/broadcast_to_all_gangs(var/message)
...
+			tag.dir = S.tagging_direction
+			if (gang)
+				gang.do_vandalism(GANG_VANDALISM_PER_GRAFFITI_TILE, S.graffititargets[1])
+		else
+			var/list/turf/turfs_ordered = new/list(length(S.graffititargets))
+			var/spraydirection = dir_to_angle(S.tagging_direction)
+			var/vec = angle_to_vector(spraydirection)
+			var/min_distance = 1000
+			for (var/i = 1; i<=targets; i++)
+				min_distance = min(vec[2] * S.graffititargets[i].y - vec[1] * S.graffititargets[i].x , min_distance)
```
Comment (FlameArrow57) [inline code/datums/gamemodes/gangwar.dm:1418]: Could probably use some comments on the math here, or algorithm?

File: code/datums/gamemodes/gangwar.dm [hunk 13]
```diff
@@ -1115,6 +1303,150 @@ proc/broadcast_to_all_gangs(var/message)
...
+				tag.dir = S.tagging_direction
+				vandal_score += GANG_VANDALISM_PER_GRAFFITI_TILE
+
+				if (gang && !chosenTurf)
+					for (var/area/targetArea as anything in gang.vandalism_tracker)
+						if (istype(area,targetArea))
+							chosenTurf = turfs_ordered[i]
+							break
+			if (gang)
+				gang.do_vandalism(vandal_score, chosenTurf)
```
Comment (FlameArrow57) [inline code/datums/gamemodes/gangwar.dm:1439]:
```suggestion
			gang?.do_vandalism(vandal_score, chosenTurf)
```

File: code/datums/gamemodes/gangwar.dm [hunk 14]
```diff
@@ -1115,6 +1303,150 @@ proc/broadcast_to_all_gangs(var/message)
...
+
+
+		S.clear_targets()
+		playsound(S.loc, 'sound/effects/graffiti_hit.ogg', 10, TRUE)
```
Comment (FlameArrow57) [inline code/datums/gamemodes/gangwar.dm:1443]: Volume 10?

File: code/datums/gamemodes/gangwar.dm [hunk 15]
```diff
@@ -1743,6 +2101,15 @@ proc/broadcast_to_all_gangs(var/message)
 		There is additional demand for [GANG_WEED_LIMIT-gang_weed] leaves of cannabis, for 10 points each."}
 		boutput(user, SPAN_ALERT(text))
 
+	proc/print_scoreboard(var/mob/living/carbon/human/user)
```
Comment (FlameArrow57) [inline code/datums/gamemodes/gangwar.dm:2104]:
```suggestion
	proc/print_scoreboard(mob/living/carbon/human/user)
```

File: code/datums/gamemodes/gangwar.dm [hunk 16]
```diff
@@ -1743,6 +2101,15 @@ proc/broadcast_to_all_gangs(var/message)
...
+		var/text = {"The current scores are:</br>"}
+		var/list/datum/gang/scores
+		var/datum/game_mode/gang/gamemode = ticker.mode
+		scores = sortListCopy(gamemode.gangs, /proc/cmp_gang_score_desc)
```
Comment (FlameArrow57) [inline code/datums/gamemodes/gangwar.dm:2108]:
```suggestion
		scores = sortListCopy(gamemode.gangs, GLOBAL_PROC_REF(cmp_gang_score_desc))
```

File: code/modules/items/specials/ItemSpecials.dm [hunk 17]
```diff
@@ -1849,6 +1860,86 @@ ABSTRACT_TYPE(/datum/item_special/spark)
 				playsound(master, 'sound/effects/swoosh.ogg', 50, FALSE)
 		return
 
+
+/datum/item_special/graffiti
+	cooldown = 30 //30
```
Comment (FlameArrow57) [inline code/modules/items/specials/ItemSpecials.dm:1865]:
```suggestion
	cooldown = 3 SECONDS
```

File: code/modules/items/specials/ItemSpecials.dm [hunk 18]
```diff
@@ -1849,6 +1860,86 @@ ABSTRACT_TYPE(/datum/item_special/spark)
...
+	desc = "Use a burst of spray paint to harmlessly blind, deface & disorient an opponent."
+	var/spray_color
+	/// If true, the swipe will ignite stuff in it's reach.
+	var/ignition = FALSE
+	var/flipped = FALSE
+
+	pixelaction(atom/target, list/params, mob/user, reach)
+		if(!isturf(target.loc) && !isturf(target)) return
+		if(!usable(user)) return
+		if(params["left"] && master && get_dist_pixel_squared(user, target, params) > ITEMSPECIAL_PIXELDIST_SQUARED)
```
Comment (FlameArrow57) [inline code/modules/items/specials/ItemSpecials.dm:1883]: Could probably do an early return here to remove need of indenting rest of function?

File: code/modules/items/specials/ItemSpecials.dm [hunk 19]
```diff
@@ -1849,6 +1860,86 @@ ABSTRACT_TYPE(/datum/item_special/spark)
...
+
+						hit_twitch(A)
+						if (ishuman(A) && prob(40))
+							var/mob/living/carbon/human/victim = A
+							victim.emote("cough")
+						attacked += A
+
+			afterUse(user)
+			playsound(master, 'sound/items/graffitispray3.ogg', 50, TRUE)
+		return
```
Comment (FlameArrow57) [inline code/modules/items/specials/ItemSpecials.dm:1941]:
```suggestion
```

File: code/modules/items/specials/ItemSpecials.dm [hunk 20]
```diff
@@ -2227,3 +2358,155 @@ ABSTRACT_TYPE(/datum/item_special/spark)
...
+			preUse(user)
+			current_chain = 0
+			var/direction = get_dir_pixel(user, target, params)
+			var/turf/turf = get_step(master, direction)
+
+			var/obj/itemspecialeffect/cleave/S = new /obj/itemspecialeffect/cleave
+			S.set_dir(direction)
+			S.setup(turf)
+			alternate = TRUE
+			var/hit = 0
```
Comment (FlameArrow57) [inline code/modules/items/specials/ItemSpecials.dm:2408]:
```suggestion
			var/hit = FALSE
```

File: code/modules/items/specials/ItemSpecials.dm [hunk 21]
```diff
@@ -2227,3 +2358,155 @@ ABSTRACT_TYPE(/datum/item_special/spark)
...
+			for(var/atom/A in turf)
+				if(isTarget(A))
+					A.Attackby(master, user, params, 1)
+					hit = 1
```
Comment (FlameArrow57) [inline code/modules/items/specials/ItemSpecials.dm:2412]:
```suggestion
					hit = TRUE
```

File: code/modules/items/specials/ItemSpecials.dm [hunk 22]
```diff
@@ -2227,3 +2358,155 @@ ABSTRACT_TYPE(/datum/item_special/spark)
...
+			var/turf/turf = get_step(master, direction)
+
+			var/obj/itemspecialeffect/cleave/S = new /obj/itemspecialeffect/cleave
+			S.set_dir(direction)
+			S.setup(turf)
+			alternate = TRUE
+			var/hit = 0
+			for(var/atom/A in turf)
+				if(isTarget(A))
+					A.Attackby(master, user, params, 1)
```
Comment (FlameArrow57) [inline code/modules/items/specials/ItemSpecials.dm:2411]:
```suggestion
					A.Attackby(master, user, params, TRUE)
```

File: code/modules/status_system/statusEffects.dm [hunk 23]
```diff
@@ -2751,3 +2751,53 @@
 			var/mob/living/M = src.owner
 			M.bioHolder.RemoveEffectInstance(src.added_accent)
 		UnregisterSignal(src.owner, COMSIG_MOB_SAY)
+
+/datum/statusEffect/graffiti
+	id = "graffiti"
```
Comment (FlameArrow57) [inline code/modules/status_system/statusEffects.dm:2756]: Could probably change id to something more descriptive than "graffiti"? Considering this is used on mobs, specifically for blocking their vision

File: code/modules/status_system/statusEffects.dm [hunk 24]
```diff
@@ -2751,3 +2751,53 @@
...
+	unique = 1
+	maxDuration = 15 SECONDS
+	var/counter = 0
+	var/sound = 'sound/effects/electric_shock_short.ogg'
+	var/count = 7
+	var/list/tag_images = list()
+	var/list/tag_filters = list()
+	movement_modifier = /datum/movement_modifier/tagged
+	effect_quality = STATUS_QUALITY_NEGATIVE
+	var/datum/hud/vision_impair_tag/hud = new
```
Comment (FlameArrow57) [inline code/modules/status_system/statusEffects.dm:2769]: Can qdel and set to null in OnRemove() I think, for proper GC

File: code/modules/items/specials/ItemSpecials.dm [hunk 25]
```diff
@@ -2227,3 +2358,155 @@ ABSTRACT_TYPE(/datum/item_special/spark)
...
+						break
+				if (!hit)
+					playsound(user, 'sound/impact_sounds/Generic_Swing_1.ogg', 40, FALSE, 0.1, 1.4)
+
+			if (current_chain > 3 && current_chain < max_chain && penalty_disorient) // penalise getting interrupted after the third swing
+				var/string ="[H] swings their machete too hard and loses their balance!"
+				H.show_message(SPAN_ALERT(string), 1)
+				H.do_disorient(H.get_stamina(), 0, 0, 0, 4 SECONDS, FALSE)
+			afterUse(user)
+		return
```
Comment (FlameArrow57) [inline code/modules/items/specials/ItemSpecials.dm:2466]:
```suggestion
```

File: code/modules/items/specials/ItemSpecials.dm [hunk 26]
```diff
@@ -2227,3 +2358,155 @@ ABSTRACT_TYPE(/datum/item_special/spark)
...
+				SS.set_dir(direction)
+				SS.setup(turf)
+				if (prob(10) && current_chain > 3)
+					user.emote(pick("laugh","cackle","grin"))
+
+				hit = 0
+				for(var/atom/A in turf)
+					if(isTarget(A))
+						A.Attackby(master, user, params, 1)
+						hit = 1
```
Comment (FlameArrow57) [inline code/modules/items/specials/ItemSpecials.dm:2456]:
```suggestion
				hit = FALSE
				for(var/atom/A as anything in turf)
					if(isTarget(A))
						A.Attackby(master, user, params, TRUE)
						hit = TRUE
```

File: code/modules/items/specials/ItemSpecials.dm [hunk 27]
```diff
@@ -2227,3 +2358,155 @@ ABSTRACT_TYPE(/datum/item_special/spark)
...
+					var/string ="[H] twirls the machete with a grin."
+					H.show_message(SPAN_ALERT(string), 1, assoc_maptext = make_chat_maptext(H, "<I>[string]</I>", "color: #C2BEBE;", alpha = 140))
+					master.on_spin_emote(user)
+					sleep(1.4 SECONDS)
+					damageMult = damage_mult_start+(current_chain*damage_mult_increment)
+				else if (current_chain < 3)
+					damageMult = damage_mult_start+(current_chain*damage_mult_increment)
+					sleep(rand(7,9))
+				else
+					sleep(rand(4,6))
```
Comment (FlameArrow57) [inline code/modules/items/specials/ItemSpecials.dm:2439]: Seconds define?

File: code/modules/items/specials/ItemSpecials.dm [hunk 28]
```diff
@@ -2227,3 +2358,155 @@ ABSTRACT_TYPE(/datum/item_special/spark)
...
+				turf = get_step(master, direction)
+				var/obj/itemspecialeffect/SS
```
Comment (FlameArrow57) [inline code/modules/items/specials/ItemSpecials.dm:2441]: Variable name other than "SS" would be best

File: code/modules/items/specials/ItemSpecials.dm [hunk 29]
```diff
@@ -2227,3 +2358,155 @@ ABSTRACT_TYPE(/datum/item_special/spark)
...
+			overrideStaminaDamage = master.stamina_damage * 0.8
+		return
+
+	pixelaction(atom/target, params, mob/user, reach)
+		damageMult = damage_mult_start
+		if(!isturf(target.loc) && !isturf(target)) return
+		if(!usable(user) || !isliving(user)) return
+
+		var/mob/living/H = user
+		if(params["left"] && master && get_dist_pixel_squared(user, target, params) > ITEMSPECIAL_PIXELDIST_SQUARED)
```
Comment (FlameArrow57) [inline code/modules/items/specials/ItemSpecials.dm:2398]: Early return?

File: code/modules/items/specials/ItemSpecials.dm [hunk 30]
```diff
@@ -2227,3 +2358,155 @@ ABSTRACT_TYPE(/datum/item_special/spark)
...
+			preUse(user)
+			current_chain = 0
+			var/direction = get_dir_pixel(user, target, params)
+			var/turf/turf = get_step(master, direction)
+
+			var/obj/itemspecialeffect/cleave/S = new /obj/itemspecialeffect/cleave
```
Comment (FlameArrow57) [inline code/modules/items/specials/ItemSpecials.dm:2404]: Variable name other than "S" would be best

File: code/datums/gamemodes/gangwar.dm [hunk 31]
```diff
@@ -1115,6 +1303,150 @@ proc/broadcast_to_all_gangs(var/message)
...
+			for (var/i = 1; i<=targets; i++)
+				min_distance = min(vec[2] * S.graffititargets[i].y - vec[1] * S.graffititargets[i].x , min_distance)
+
+			for (var/sorting = 1; sorting<=targets; sorting++)
+				var/dist = (vec[2] * S.graffititargets[sorting].y -vec[1] * S.graffititargets[sorting].x ) - min_distance
+				turfs_ordered[dist+1] = S.graffititargets[sorting]
+
+			var/vandal_score = 0
+			var/turf/chosenTurf
+			for (var/i = 1; i<=targets; i++)
```
Comment (FlameArrow57) [inline code/datums/gamemodes/gangwar.dm:1426]:
```suggestion
			for (var/i = 1 to targets)
```

**Review by FlameArrow57 — COMMENTED** [review]
Cool! Just did a minor styling/sanity check review, not in-depth

File: code/modules/status_system/statusEffects.dm (hunk 4 above)
Comment (FlameArrow57) [reply code/modules/status_system/statusEffects.dm:2796]: image

File: _std/defines/gang.dm [hunk 32]
```diff
@@ -133,11 +135,31 @@
...
+/// How many points throwing a person into a vending machine is worth
+#define GANG_VANDALISM_VENDOR_KO 10
+/// How many points throwing a person into a glass table
+#define GANG_VANDALISM_TABLING 10
+/// How many points ripped up floor tiles are worth
+#define GANG_VANDALISM_FLOORTILE_POINTS 2
+/// How many points each point of damage is worth, for violence
+#define GANG_VANDALISM_VIOLENCE_NPC_MULTIPLIER 0.5
+/// How many points each point of damage is worth, for violence
+#define GANG_VANDALISM_VIOLENCE_PLAYER_MULTIPLIER 0
```
[the remark this answers is not in the document]
Comment (TDHooligan) [reply _std/defines/gang.dm:159]: it was 0.25 but then i realised the massacres will start again. it's fully implemented tho

File: code/datums/controllers/process/gang.dm [hunk 33]
```diff
@@ -132,13 +132,20 @@
...
+	var/list/unvandalised_departments = list()
 	setup()
-		name = "Gang_Duffle_Drops"
+		name = "Gang_Duffle_Objectives"
 		schedule_interval = GANG_LOOT_INITIAL_DROP
-		src.repeats = GANG_LOOT_DROP_VOLUME_PER_GANG
+		unvandalised_departments += /area/station/engine
+		unvandalised_departments += /area/station/medical
+		unvandalised_departments += /area/station/quartermaster
+		//unvandalised_departments += /area/station/janitor
```
[the remark this answers is not in the document]
Comment (FlameArrow57) [reply code/datums/controllers/process/gang.dm:145]: There's an uncommented line with no context, would suggest removing it if not needed, or adding context

File: code/datums/gamemodes/gangwar.dm [hunk 34]
```diff
@@ -1115,6 +1303,150 @@ proc/broadcast_to_all_gangs(var/message)
...
+	interrupt_flags = INTERRUPT_STUNNED
+	icon = 'icons/obj/items/items.dmi'
+	icon_state = "spraycan"
+	var/obj/item/spray_paint_graffiti/S
+	/// the mob spraying this tag
+	var/mob/M
+	/// the gang we're spraying for
+	var/datum/gang/gang
+	/// when our next spray sound can beplayed
+	var/next_spray = 0 DECI SECONDS
```
[the remark this answers is not in the document]
Comment (TDHooligan) [reply code/datums/gamemodes/gangwar.dm:1318]: spray sounds start at 0 so the first one plays

File: code/datums/gamemodes/gangwar.dm (hunk 14 above)
Comment (TDHooligan) [reply code/datums/gamemodes/gangwar.dm:1443]: sfx is loud

File: code/procs/mobprocs/attacks.dm [hunk 35]
```diff
@@ -30,10 +30,18 @@
 
 	if (!shielded || !(W.flags & NOSHIELD))
 		// drsingh Cannot read null.force
-#ifdef DATALOGGER
 		if (W.force)
+#ifdef DATALOGGER
```
[the remark this answers is not in the document]
Comment (TDHooligan) [reply code/procs/mobprocs/attacks.dm:34]: scopes datalogger define to the violence increment instead of everything related to force

File: code/datums/gamemodes/gangwar.dm (hunk 9 above)
Comment (TDHooligan) [reply code/datums/gamemodes/gangwar.dm]: initial size.

**Comment by TDHooligan** [comment]
merging caused buggery.

**Comment by NightmareChamillian** [comment]
I love this! Can I suggest breaking (or maybe only damaging to prevent depressurization) windows as well?

**Comment by TDHooligan** [comment]
> I love this! Can I suggest breaking (or maybe only damaging to prevent depressurization) windows as well?

already considered but if you're told smack a window and get points for it, odds are youre not gonna stop just because you arent getting number for it any more

File: code/obj/item/misc_weapons.dm [hunk 36]
```diff
@@ -2207,3 +2211,34 @@ obj/item/whetstone
 	New()
 		..()
 		src.setItemSpecial(/datum/item_special/rangestab)
+
+
+
+
+/obj/item/gang_machete
```
Comment (ZeWaka) [inline code/obj/item/misc_weapons.dm:2218]: please put this in an organized file, not dump it here

File: code/modules/items/specials/ItemSpecials.dm [hunk 37]
```diff
@@ -2232,3 +2363,153 @@ ABSTRACT_TYPE(/datum/item_special/spark)
 		if(progress == 1)
 			state = ACTIONSTATE_FINISH
 			return
+
+
+/datum/item_special/massacre
```
Comment (ZeWaka) [inline code/modules/items/specials/ItemSpecials.dm:2368]: ditto

File: code/modules/items/specials/ItemSpecials.dm [hunk 38]
```diff
@@ -1853,6 +1864,86 @@ ABSTRACT_TYPE(/datum/item_special/spark)
 				playsound(master, 'sound/effects/swoosh.ogg', 50, FALSE)
 		return
 
+
+/datum/item_special/graffiti
```
Comment (ZeWaka) [inline code/modules/items/specials/ItemSpecials.dm:1868]: ditto

**Review by ZeWaka — CHANGES_REQUESTED** [review]
tldr organize please

File: code/modules/items/specials/ItemSpecials.dm [hunk 39]
```diff
@@ -1853,6 +1864,86 @@ ABSTRACT_TYPE(/datum/item_special/spark)
...
+		flipped = !flipped
+		graffiti.color = pick("#FF0000","#FF9A00","#FFFF00","#00FF78","#00FFFF","#0081DF","#CC00FF","#FFCCFF","#EBE6EB")
+
+		graffiti.setup(effect)
+		graffiti.set_dir(direction)
+
+		for(var/turf/T in list(one, two, three))
+			for(var/atom/movable/A in T)
+				if(A in attacked) continue
+				if(isliving(A))
```
Comment (TobleroneSwordfish) [inline code/modules/items/specials/ItemSpecials.dm:1914]: Make this a `continue` guard too, cleaner

File: code/modules/status_system/statusEffects.dm [hunk 40]
```diff
@@ -2797,6 +2797,60 @@
...
+/datum/statusEffect/graffiti
+	id = "graffiti_blind"
+	name = "Tagged!"
+	desc = "You've been tagged! <br>Movement speed is reduced. Eyesight reduced. "
+	icon_state = "tagged"
+	unique = TRUE
+	maxDuration = 15 SECONDS
+	var/counter = 0
+	var/sound = 'sound/effects/electric_shock_short.ogg'
+	var/count = 7
```
Comment (TobleroneSwordfish) [inline code/modules/status_system/statusEffects.dm:2809]: Doc comment `counter` and `count`, easily confused var intent

File: code/modules/status_system/statusEffects.dm [hunk 41]
```diff
@@ -2797,6 +2797,60 @@
...
+			if (prob(10) && ismob(owner))
+				var/mob/victim = owner
+				victim.emote(pick("cough", "blink"))
+			playsound(owner, sound, 17, TRUE, 0.4, 1.6)
+			violent_twitch(owner)
+		. = ..(timePassed)
+	onRemove()
+		qdel(hud)
+		hud = null
+		. = ..()
```
Comment (TobleroneSwordfish) [inline code/modules/status_system/statusEffects.dm:2852]: Double `onRemove` definition (don't you just love that this is legal byond code?)

File: code/obj/item/misc_weapons.dm [hunk 42]
```diff
@@ -2207,3 +2211,34 @@ obj/item/whetstone
...
+	throw_range = 5
+	contraband = 4
+	w_class = W_CLASS_NORMAL
+	tool_flags = TOOL_CUTTING
+	attack_verbs = "slashes"
+	hitsound = 'sound/impact_sounds/Blade_Small_Bloody.ogg'
+
+	New()
+		..()
+		src.AddComponent(/datum/component/bloodflick)
```
Comment (TobleroneSwordfish) [inline code/obj/item/misc_weapons.dm:2240]: Not sure if this an issue with the base component but this seems to cause a runtime on the third attack of each chain
image

File: code/obj/item/misc_weapons.dm [hunk 43]
```diff
@@ -2207,3 +2211,34 @@ obj/item/whetstone
...
+		src.setItemSpecial(/datum/item_special/massacre)
+	on_spin_emote(mob/user)
+		SEND_SIGNAL(src, COMSIG_ITEM_TWIRLED, user, src)
+		animate_spin(src, prob(50) ? "L" : "R", 1, 0)
```
Comment (TobleroneSwordfish) [inline code/obj/item/misc_weapons.dm:2244]: What is this actually doing? It just seems to be recreating some of the base behaviour of the `twirl` emote (and sending the signal twice). If you want the behaviour of the twirl emote then just make the user do the emote instead.

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]
I absolutely love the new graffiti stuff, it feels great to use and the sprites look incredible!
But for the love of god in future please atomize these changes 🥺 this could have been at least 3 separate PRs very easily and would likely have gotten reviewed/merged a lot faster!

**Comment by Cherman0** [comment]
Reopened by request

File: code/datums/gamemodes/gangwar.dm [hunk 44]
```diff
@@ -2106,53 +2462,73 @@ proc/broadcast_to_all_gangs(var/message)
...
-	class1 = "Misc Gang"
-	class1 = "Consumable"
+	category = "Weapon"
+/datum/gang_item/equipment
+	category = "Consumable"
+
+/datum/gang_item/equipment/graffiti
+	name = "'ProPaint' Spray Can"
+	desc = "Non-permanent graffiti, great for vandalism & blinding the fuzz. Not able to claim territory."
+	class2 = "consumable"
```
[the remark this answers is not in the document]
Comment (TDHooligan) [reply code/datums/gamemodes/gangwar.dm:2490]: more equipment was added, removed pending adding later in a seperate PR

File: code/obj/item/misc_weapons.dm (hunk 42 above)
Comment (TDHooligan) [reply code/obj/item/misc_weapons.dm]: base class error - idk exactly waht the fix is atm.

Comment (TDHooligan) [reply code/obj/item/misc_weapons.dm]: probably visible_message

**Comment by TobleroneSwordfish** [comment]
You know what, we fix it live.
